Mathlib/Algebra/Homology/ShortComplex/SnakeLemma.lean

```diff
@@ -0,0 +1,204 @@
+  /-- the morphism from the zeroth row to the first row -/
+  v₀₁ : L₀ ⟶ L₁
+  /-- the morphism from the first row to the second row -/
+  v₁₂ : L₁ ⟶ L₂
+  /-- the morphism from the second row to the third row -/
+  v₂₃ : L₂ ⟶ L₃
+  w₀₂ : v₀₁ ≫ v₁₂ = 0 := by aesop_cat
+  w₁₃ : v₁₂ ≫ v₂₃ = 0 := by aesop_cat
+  /-- `L₀` is the kernel of `v₁₂ : L₁ ⟶ L₂`. -/
+  h₀ : IsLimit (KernelFork.ofι _ w₀₂)
+  /-- `L₃` is the cokernel of `v₁₂ : L₁ ⟶ L₂`. -/
+  h₃ : IsColimit (CokernelCofork.ofπ _ w₁₃)
+  L₁_exact  : L₁.Exact
+  epi_L₁_g : Epi L₁.g
+  L₂_exact : L₂.Exact
+  mono_L₂_f : Mono L₂.f
+
+initialize_simps_projections SnakeInput (-h₀, -h₃)
+
+namespace SnakeInput
+
+attribute [reassoc (attr := simp)] w₀₂ w₁₃
+attribute [instance] epi_L₁_g
+attribute [instance] mono_L₂_f
+
+variable {C}
+variable (S : SnakeInput C)
+
+/-- The snake input in the opposite category that is deduced from a snake input. -/
+@[simps]
+noncomputable def op : SnakeInput Cᵒᵖ where
+  L₀ := S.L₃.op
+  L₁ := S.L₂.op
+  L₂ := S.L₁.op
+  L₃ := S.L₀.op
+  epi_L₁_g := by dsimp; infer_instance
+  mono_L₂_f := by dsimp; infer_instance
+  v₀₁ := opMap S.v₂₃
+  v₁₂ := opMap S.v₁₂
+  v₂₃ := opMap S.v₀₁
+  w₀₂ := congr_arg opMap S.w₁₃
+  w₁₃ := congr_arg opMap S.w₀₂
+  h₀ := isLimitForkMapOfIsLimit' (ShortComplex.opEquiv C).functor _
+      (CokernelCofork.IsColimit.ofπOp _ _ S.h₃)
+  h₃ := isColimitCoforkMapOfIsColimit' (ShortComplex.opEquiv C).functor _
+      (KernelFork.IsLimit.ofιOp _ _ S.h₀)
+  L₁_exact := S.L₂_exact.op
+  L₂_exact := S.L₁_exact.op
+
+@[reassoc (attr := simp)] lemma w₀₂_τ₁ : S.v₀₁.τ₁ ≫ S.v₁₂.τ₁ = 0 := by
+  rw [← comp_τ₁, S.w₀₂, zero_τ₁]
+@[reassoc (attr := simp)] lemma w₀₂_τ₂ : S.v₀₁.τ₂ ≫ S.v₁₂.τ₂ = 0 := by
+  rw [← comp_τ₂, S.w₀₂, zero_τ₂]
+@[reassoc (attr := simp)] lemma w₀₂_τ₃ : S.v₀₁.τ₃ ≫ S.v₁₂.τ₃ = 0 := by
+  rw [← comp_τ₃, S.w₀₂, zero_τ₃]
+@[reassoc (attr := simp)] lemma w₁₃_τ₁ : S.v₁₂.τ₁ ≫ S.v₂₃.τ₁ = 0 := by
+  rw [← comp_τ₁, S.w₁₃, zero_τ₁]
+@[reassoc (attr := simp)] lemma w₁₃_τ₂ : S.v₁₂.τ₂ ≫ S.v₂₃.τ₂ = 0 := by
+  rw [← comp_τ₂, S.w₁₃, zero_τ₂]
+@[reassoc (attr := simp)] lemma w₁₃_τ₃ : S.v₁₂.τ₃ ≫ S.v₂₃.τ₃ = 0 := by
+  rw [← comp_τ₃, S.w₁₃, zero_τ₃]
+
+/-- `L₀.X₁` is the kernel of `v₁₂.τ₁ : L₁.X₁ ⟶ L₂.X₁`. -/
+noncomputable def h₀τ₁ : IsLimit (KernelFork.ofι S.v₀₁.τ₁ S.w₀₂_τ₁) :=
+  isLimitForkMapOfIsLimit' π₁ S.w₀₂ S.h₀
+
+/-- `L₀.X₂` is the kernel of `v₁₂.τ₂ : L₁.X₂ ⟶ L₂.X₂`. -/
+noncomputable def h₀τ₂ : IsLimit (KernelFork.ofι S.v₀₁.τ₂ S.w₀₂_τ₂) :=
+  isLimitForkMapOfIsLimit' π₂ S.w₀₂ S.h₀
+
+/-- `L₀.X₃` is the kernel of `v₁₂.τ₃ : L₁.X₃ ⟶ L₂.X₃`. -/
+noncomputable def h₀τ₃ : IsLimit (KernelFork.ofι S.v₀₁.τ₃ S.w₀₂_τ₃) :=
+  isLimitForkMapOfIsLimit' π₃ S.w₀₂ S.h₀
+
+instance mono_v₀₁_τ₁ : Mono S.v₀₁.τ₁ := mono_of_isLimit_fork S.h₀τ₁
+instance mono_v₀₁_τ₂ : Mono S.v₀₁.τ₂ := mono_of_isLimit_fork S.h₀τ₂
+instance mono_v₀₁_τ₃ : Mono S.v₀₁.τ₃ := mono_of_isLimit_fork S.h₀τ₃
+
+/-! The upper part of the first column of the snake diagram is exact. -/
```

Comment:
```suggestion
/-- The upper part of the first column of the snake diagram is exact. -/
```
I don't know if `/-!` does something, but I think this is a typo. Similar elsewhere.

Reply:
Thanks!